aws-datastore/src/main/java/com/amplifyframework/datastore/AWSDataStorePlugin.java

```diff
@@ -185,12 +189,16 @@ public void configure(
 
     @WorkerThread
     @Override
-    public void initialize(@NonNull Context context) {
-        Completable completable = initializeStorageAdapter(context);
-        if (!api.getPlugins().isEmpty()) {
-            completable = completable.andThen(orchestrator.start());
+    public void initialize(@NonNull Context context) throws AmplifyException {
+        Throwable initError = initializeStorageAdapter(context)
+            .andThen(initializeOrchestrator())
+            .blockingGet(PLUGIN_INIT_TIMEOUT_MS, TimeUnit.MILLISECONDS);
+        if (initError != null) {
+            throw new AmplifyException("Failed to initialize the local storage " +
```

Comment:
Nit: maybe newline instead of concat?

Reply:
Why is this more likely to happen after clear?

Reply:
@jpignata I don't think its more likely. I have this code in one of my PRs, too, though. Probably, we could carve this out into its own targeted PR. (Not saying we should, at this point, just that it would be possible.)

@rjuliano and I have been talking offline about reducing/removing all of our dangerous `blockPermanently()` type calls, which can hose threads. We only have a few such calls in source code, but there are quite a few in the tests, right now. We should start whittling away at them by providing some sane timeouts, so that nothing is wedged with permanence.